integration_tests/tests/e2e-tests.ts

```diff
@@ -0,0 +1,154 @@
+    async (data) => {
+      eventIds = await generateTransactionEventsViaFilterLambda(
+        data.eventTime,
+        data.transactionQty,
+        eventName
+      );
+      eventTime = data.eventTime;
+
+      invoiceFileName = await createInvoiceWithGivenData(
+        data,
+        dataRetrievedFromConfig.description,
+        dataRetrievedFromConfig.unitPrice,
+        dataRetrievedFromConfig.vendorId,
+        dataRetrievedFromConfig.vendorName
+      );
+      await assertQueryResultWithTestData(
+        data,
+        dataRetrievedFromConfig.unitPrice,
+        dataRetrievedFromConfig.vendorId,
+        dataRetrievedFromConfig.serviceName
+      );
+    }
+  );
+
+  test.each`
+    testCase                                                                                 | eventTime       | transactionQty | billingQty
+    ${"No BillingQty No Billing Price (no invoice) but has TransactionQty TransactionPrice"} | ${"2023/02/28"} | ${"1"}         | ${undefined}
+  `(
+    "results retrieved from billing and transaction_curated view query should match with expected $testCase,$eventTime,$unitPrice,$transactionQty,$billingQty,$transactionPrice,$billingPrice,$priceDiff,$qtyDiff,$priceDifferencePercent,$qtyDifferencePercent",
+    async (data) => {
+      eventIds = await generateTransactionEventsViaFilterLambda(
+        data.eventTime,
+        data.transactionQty,
+        eventName
+      );
+      eventTime = data.eventTime;
+      await assertQueryResultWithTestData(
+        data,
+        dataRetrievedFromConfig.unitPrice,
+        dataRetrievedFromConfig.vendorId,
+        dataRetrievedFromConfig.serviceName
+      );
+    }
+  );
+
+  afterEach(async () => {
+    await deleteS3Events(eventIds, eventTime);
+    await deleteS3Object({
+      bucket: storageBucket,
+      key: `${standardisedFolderPrefix}/${invoiceFileName.slice(0, 27)}.txt`,
+    });
+  });
+});
+
+export const assertQueryResultWithTestData = async (
+  { billingQty, transactionQty, eventTime }: TestData,
+  unitPrice: number,
+  vendorId: string,
+  serviceName: string
+): Promise<void> => {
+  const tableName = TableNames.BILLING_TRANSACTION_CURATED;
+  const response: BillingTransactionCurated =
+    await queryResponseFilterByVendorServiceNameYearMonth(
+      vendorId,
+      serviceName,
+      tableName,
+      eventTime
+    );
+
+  const billingPrice = unitPrice * billingQty;
+  const transactionPrice = unitPrice * transactionQty;
+  const qtyDifference = billingQty - transactionQty;
+  const priceDifference = billingPrice - transactionPrice;
+  const priceDifferencePercentage = (priceDifference / transactionPrice) * 100;
+  const qtyDifferencePercentage = (qtyDifference / transactionQty) * 100;
+
+  if (billingQty !== undefined && transactionQty !== undefined) {
```

Comment:
I think the fact you've had to do this branching here is a clue that maybe these cases don't fit neatly into a `test.each` style

Reply:
after addingCalculateExpectedResults function the branching looks clean and because of  test.each expect statement wont be repeated

Reply:
@Si3rr4wow Mark is fine with change and agreed to resolve